src/Type/ClassStringType.php

```diff
@@ -44,7 +44,7 @@ public function accepts(Type $type, bool $strictTypes): TrinaryLogic
 	public function isSuperTypeOf(Type $type): TrinaryLogic
 	{
 		if ($type instanceof ConstantStringType) {
-			return TrinaryLogic::createFromBoolean($type->isClassString());
+			return TrinaryLogic::createFromBoolean($type->isClassStringType()->yes());
```

Comment:
You can simplify this and return the original trinary here and below maybe?

Reply:
I thought so too, but I think the alternative would be `$type->isClassStringType()->and(Trinary::createYes())` which is not that readable either.

Reply:
see https://github.com/phpstan/phpstan-src/pull/1970/files#r1014798873 ;)